docs/PREACT_BADGES.md

```diff
@@ -0,0 +1,328 @@
+
+#### The following badge template properties are **optional**:
+
+`value.enabled` - boolean that when true, required a badge `value` to be provided when using this template
+
+`value.validations.min` - ensures `value` meets a numerical minimum or `string` / `url` length
+
+`value.validations.max` - ensures `value` meets a numerical maximum or `string` / `url` length
+
+`value.validations.regex` - ensures `value` meets a regex definition. Must also provide `value.validations.regexExplain`
+
+`value.validations.regexExplain` - required if using `value.validations.regex`. Describes the regex definition and is displayed as an error message if the regex validation fails
+
+
+### Badge Template Locations
+Badge template locations is an array of strings. 
+
+Possible values when using default locations: `left`, `left/left`, `right`, `right/right`, `callout`, `callout/callout`
+
+Possible values when using **custom** locations: `left`, `left/[tag]`, `right`, `right/[tag]`, `callout`, `callout/[tag]`. See `Custom Badge Locations` section below for creating custom locations.
+
+For example, if the locations.json file contains the following location definition:
+
+```json
+{
+	"left": [
+		{
+			"tag": "left",
+			"name": "Top Left"
+		},
+		{
+			"tag": "left-bottom",
+			"name": "Bottom Left"
+		}
+	]
+}
+```
+
+To restrict a badge template to a custom location, the badge template `locations` array should contain the `tag` of the locations. Ie. `left/left-bottom`
+
+
+### Badge Template Parameters
+Badge template parameters is an array of objects. Each object is a template parameter and contains the following properties: 
+
+#### The following badge template parameters properties are **required**:
+
+`name` - unique badge location identifier
+
+`type` - parameter value type. Available types: `array`, `string`, `color`, `url`, `integer`, `decimal`, `boolean`, `checkbox`, `toggle`. See example below for example usage of each type. 
+
+`label` - label that is displayed when selecting this badge location within the Searchspring Management Console
+
+`description` - badge location description
+
+`options` - required only if `type` is `array`. Defines a list of available values to select.
+
+
+#### The following badge template parameters properties are **optional**:
+
+`defaultValue` - default value that will be used unless specified when configuring a new badge rule. Must be a string regardless of different `type` options.
+
+`validations` - only applicable if `type` is `string`, `url`. `integer`, `decimal`
+
+`validations.min` - only applicable if `type` is `integer` or `decimal`. Ensures `defaultValue` or the user defined `value` meets a numerical minimum or `string` / `url` length
+
+`validations.max` - only applicable if `type` is `integer` or `decimal`. Ensures `defaultValue` or the user defined `value` meets a numerical maximum or `string` / `url` length
```

Comment:
Contradictory information here, says only when type is `integer` or `decimal`, but mentions `string` and `url` immediately after.